cmd/yggdrasil/main.go

```diff
@@ -271,15 +272,53 @@ func main() {
 			n.tun.SetupAdminHandlers(n.admin)
 		}
 	}
+	
+	//Windows service shutdown service
+	minwinsvc.SetOnExit(func() {
+		logger.Infof("Shutting down service ...")
+		sigCh <- os.Interrupt
+		// Wait for all parts to shutdown properly
+		<-doneCh
+	})
 
 	// Block until we are told to shut down.
-	<-ctx.Done()
+	<-sigCh
+
+	// Shut down the node using a wait group to synchronize
+	var wg sync.WaitGroup
```

Comment:
Bit confused about what all this achieves, as all three `Stop()` functions block until that module is shut down anyway and therefore run sequentially.

This also just means that potentially, depending on scheduling, that `core.Stop()` can run before the others do which doesn't seem desirable.